continuationsPlugin/src/it/scala/continuations/NonCompanionObjectSnapshotsSuite.scala

```diff
@@ -0,0 +1,465 @@
+package continuations
+
+import dotty.tools.dotc.core.Contexts.Context
+import munit.SnapshotSuite
+
+class NonCompanionObjectSnapshotsSuite extends SnapshotSuite, CompilerFixtures:
+
+  compilerContextWithContinuationsPlugin.snapshotTest("test-1") {
+    case given Context =>
+      val source =
+        """
+          |import continuations.*
+          |
+          |def program = {
+          | println(ExampleObject.continuations(1))
+          |}
+          |
+          |object ExampleObject {
+          |  private def method1(x: Int) = x + 1
+          |  private val z1 = 1
+          |
+          |  protected def method2(x: Int) = x + 1
+          |  protected val z2 = 1
+          |
+          |  def method3(x: Int) = x + 1
+          |  val z3 = 1
+          |
+          |  def continuations(x: Int)(using s: Suspend): Int = {
+          |    s.suspendContinuation[Int] { continuation =>
+          |      continuation.resume(Right(method1(x) + method2(x) + method3(x) + z1 + z2 + z3 + 1))
+          |    }
+          |
+          |    def method4(x: Int) = x + 1
+          |    val z4 = 1
+          |
+          |    1 + method1(x) + method2(x) + method3(x) + method4(x) + z1 + z2 + z3 + z4
+          |  }
+          |}
+          |""".stripMargin
+
+      continuationsCompilerSnapshot(source)
+  }
+
+  compilerContextWithContinuationsPlugin.snapshotTest("test-2") {
+    case given Context =>
+      val source =
+        """
+          |import continuations.*
+          |
+          |def program = {
+          | println(ExampleObject.continuations(1))
+          |}
+          |
+          |object ExampleObject {
+          |  def continuations(x: Int)(using s: Suspend): Int = {
+          |    def method1(x: Int) = x + 1
+          |    val z1 = 1
+          |
+          |    s.suspendContinuation[Int] { continuation =>
+          |      def method2(x: Int) = x
+          |      val z2 = 1
+          |
+          |      continuation.resume(Right(method1(x) + method2(x) + z1 + z2 + 1))
+          |    }
+          |
+          |    def method3(x: Int) = x + 1
+          |    val z3 = 1
+          |
+          |    method3(x) + z1 + z3 + 1
+          |  }
+          |}
+          |""".stripMargin
+
+      continuationsCompilerSnapshot(source)
+  }
+
+  compilerContextWithContinuationsPlugin.snapshotTest("test-3".ignore) {
```

Comment:
tests with `ignore` contain code that fails when it runs so we need to debug them. the test itself passes because the tree seems okay but in running with `scalac` it has an error. unfortunately we need sbt in order to debug these